### PR TITLE
Add CFG to generate_scope in tutorials

### PR DESCRIPTION
**Pull request by ThrudPrimrose**
The DFG is missing from codegen tutorial, resulting with an error in codegen. Adding cfg as a param fixes it.

### CODE REVIEW

**Comment by ThrudPrimrose** [comment]
When I saved it also trimmed some trailing whitespace (lines that only have whitespace, it is for the better I think)

File: tutorials/codegen.ipynb [hunk 1]
```diff
@@ -480,66 +480,66 @@
...
-    "        \n",
+    "\n",
     "        # You can similarly use register_{array,copy,node,state}_dispatcher\n",
-    "        \n",
-    "    # A scope dispatcher will trigger a method called generate_scope whenever \n",
+    "\n",
+    "    # A scope dispatcher will trigger a method called generate_scope whenever\n",
     "    # an SDFG has a scope with that schedule\n",
-    "    def generate_scope(self, sdfg: dace.SDFG, scope: ScopeSubgraphView,\n",
+    "    def generate_scope(self, sdfg: dace.SDFG, dfg, scope: ScopeSubgraphView,\n",
```
Comment (tbennun) [inline tutorials/codegen.ipynb:495]: Add type hint

File: tutorials/codegen.ipynb [hunk 2]
```diff
@@ -480,66 +480,66 @@
...
     "                       state_id: int, function_stream: CodeIOStream,\n",
     "                       callsite_stream: CodeIOStream):\n",
     "        # The parameters here are:\n",
     "        # sdfg: The SDFG we are currently generating.\n",
     "        # scope: The subgraph of the state containing only the scope (map contents)\n",
```
Comment (tbennun) [inline tutorials/codegen.ipynb:500]: add tutorial documentation

**Review by tbennun — CHANGES_REQUESTED** [review]

File: tutorials/codegen.ipynb (hunk 1 above)
Comment (phschaad) [reply tutorials/codegen.ipynb:495]: Note that it should be `cfg`, not `dfg` anyway: https://github.com/spcl/dace/blob/main/dace/codegen/targets/target.py#L81

File: tutorials/codegen.ipynb [hunk 3]
```diff
@@ -480,66 +480,66 @@
...
     "            # NOTE: CodeIOStream will automatically take care of indentation for us.\n",
-    "        \n",
-    "        \n",
+    "\n",
+    "\n",
     "        # Now that the loops have been defined, use the dispatcher to invoke any\n",
     "        # code generator (including this one) that is registered to deal with\n",
     "        # the internal nodes in the subgraph. We skip the MapEntry node.\n",
-    "        self.dispatcher.dispatch_subgraph(sdfg, scope, state_id,\n",
+    "        self.dispatcher.dispatch_subgraph(sdfg, dfg, scope, state_id,\n",
```
Comment (phschaad) [inline tutorials/codegen.ipynb:539]: Same as above, the argument is `cfg`: https://github.com/spcl/dace/blob/main/dace/codegen/dispatcher.py#L380

**Review by phschaad — CHANGES_REQUESTED** [review]

File: tutorials/codegen.ipynb (hunk 2 above)
Comment (ThrudPrimrose) [reply tutorials/codegen.ipynb:502]: Added

**Pull request merged**